### ISSUE TITLE
`npm link` still create lockfile even the `package-lock` has been set to `false`

### ISSUE THREAD

**Issue by LitoMore** [npm/npm#20105]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

I set `package-lock` to `false` in `.npmrc` file, but the `npm link` command still create a lock file.

#### How can the CLI team reproduce the problem?

1. In any project, just set `package-lock` to `false` in the `.npmrc` file.
2. Run `npm link`.

### supporting information:

 - `npm -v` prints: 5.8.0-next.0
 - `node -v` prints: v9.8.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: macOS
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by LitoMore** [comment]
Is this a bug? Or it just a feature?

**Comment by serut** [comment]
I don't think this is intentional. Every npm link creates a new `package-lock.json.415xxxx` like this one : 

```
{
  "name": "@regardsoss/theme",
  "version": "0.0.1",
  "lockfileVersion": 1
}
```

**Comment by oleics** [comment]
jep, same here :-/

**Comment by miraage** [comment]
I had this issue in earlier npm versions. Currently it creates a `package-lock.json` without any suffixes with npm@5.8.0

**Comment by Jordyderuijter** [comment]
Confirmed this also happens for me. Would be nice if npm link doesn't generate the package-lock files if package-lock is set to false.
